source/PlayerInfo.cpp

```diff
@@ -1043,6 +1054,22 @@ void PlayerInfo::Land(UI *ui)
 			ui->Push(new Dialog(message));
 	}
 	
+	// Hire crew for the flagship if the user changed it from the default.
+	if(Preferences::Has("Rehire extra crew when lost") && hasSpaceport && flagshipCrew)
+	{
+		int addedCrew = flagshipCrew - flagship->Crew();
+		if(addedCrew > 0)
+		{
+			flagship->AddCrew(addedCrew);
+			Messages::Add("You hire " + Format::Number(addedCrew) + " extra crew member"
+					+ (addedCrew == 1 ? " to replace the one " : "s to replace the ones ")
+					+ "lost during boarding operations.");
+		}
+		// A new flagship was chosen with a higher crew requirement.
```

Comment:
This would mean that if the player switches to a flagship with lower crew requirements, they will automatically hire extra crew even if they didn't intend to.

I think a better approach would be to save, not the desired flagship crew, but the desired number of extra crew members.

Reply:
Won't that have the same issue?
Since the rehire action happens on landing, performing a check during PlayerInfo::TakeOff is perhaps the better solution to the player swapping from a capping ship to their "daily flier"